### PR TITLE
Test on Node.js using Jest

### PR DESCRIPTION
**Pull request by realityking**
This PR moves those tests that are not browser specific to Jest.

This also allowed cleaning up the `getWindow()` workaround to simulate a non-browser environment in a browser.

Technically this is semver-major as it removes an exported method. Since the method is not documented (not even as part of the TypeScript definition), I think it's worth the risk.

### CODE REVIEW

File: src/waypoint.jsx [hunk 1]
```diff
@@ -370,11 +372,5 @@ Waypoint.above = ABOVE;
 Waypoint.below = BELOW;
 Waypoint.inside = INSIDE;
 Waypoint.invisible = INVISIBLE;
-Waypoint.getWindow = () => {
```
Comment (lencioni) [inline src/waypoint.jsx]: It looks like this was added by 4e22e27e9290bba61f4c133509cc8e2611e72079 but I'm not really sure why. @trotzig do you happen to know?

Comment (realityking) [reply src/waypoint.jsx]: I believe it was only added to allow this mock: 

https://github.com/brigade/react-waypoint/commit/4e22e27e9290bba61f4c133509cc8e2611e72079#diff-27fadc9a7e55db698d635052018a532aR424

Comment (lencioni) [reply src/waypoint.jsx]: Cool, yeah I think this is okay to remove!

**Comment by realityking** [comment]
I've updated this PR to consider more recent changes. I think it's ready to merge.

It currently includes the changes from #332 so it could be tested. I'll remove these once that PR has landed.

**Comment by realityking** [comment]
It took a while but this PR is finally in a state where I think it's ready to go.

Note that the final commit could be considered a breaking change. I think it's very low risk since the method was never documented but let me know if you want me to replace it with a deprecation message.

**Comment by realityking** [comment]
@trotzig I have a couple of other things in mind. Can you wait ~1 week for the next release?

**Comment by trotzig** [comment]
Will do! Let me know when you're ready.

**Pull request merged**